src/js/yeldaWebchatInjector.js

```diff
@@ -775,40 +775,109 @@ class YeldaChat {
     return isFound
   }
 
+  /**
+   * Get the app env from the window url
+   * @param {String} href 
+   * @returns {String} env
+   */
+  getAppEnv(href) {
+    if (config.REGEX_HOST.STAGING.test(href)) {
+      return 'staging'
+    }
+    if (config.REGEX_HOST.LOCALHOST.test(href)) {
+      return 'local'
+    }
+    return 'prod'
+  }
+
+  /**
+   * Get the chat bubble url base based on isFallback
+   * @param {Object} data { assistantUrl, assistantId, settingId }
+   * @param {Boolean} isFallback fallback means yelda endpoint
+   * @returns {String} url base
+   */
+  getChatBubbleUrlBase(data, isFallback) {
+    if (isFallback) {
+      return `${data.assistantUrl}/assistants`
+    }
+    return 'https://webchat.yelda.ai/webchat'
+  }
+
   /**
    * return getAssistantSettings endpoint URL from data
    * @param {Object} data { assistantUrl, assistantId, settingId }
+   * @param {String} urlBase
    * @return {String} url
    */
-  getAssistantSettingsUrl(data) {
+  getAssistantSettingsUrl(data, urlBase) {
+    // The app env is required as query for the external chatbubble endpoint
+    const appEnv = this.getAppEnv(window.location.href)
+    const envQuery = `env=${appEnv}`
     if(data.settingId) {
-      return  `${data.assistantUrl}/assistants/settings/${data.settingId}/chatBubble`
+      return  `${urlBase}/settings/${data.settingId}/chatBubble?${envQuery}`
     }
 
-    return  `${data.assistantUrl}/assistants/${data.assistantId}/chatBubble/${data.locale}`
+    return  `${urlBase}/${data.assistantId}/chatBubble/${data.locale}?${envQuery}`
   }
 
   /**
    * Update assistantImage with assistant settings from backend if any
    * @param {Object} data { assistantUrl, assistantId, settingId }
    * @return {Promise}
    */
-  getAssistantSettings(data) {
+  getAssistantSettings(data, { isFallback = false, customTimeout = null } = {}) {
+    // Set `this` to be able to do some recursion in onreadystatechange
+    const self = this;
```

Comment:
:gear: 2010 code ;)
I would recommend that you recheck apply / bind / call + have a look to https://stackoverflow.com/questions/2130241/pass-correct-this-context-to-settimeout-callback